### PR TITLE
Modify rule S1244: promote C# to SonarWay

### PR DESCRIPTION
**Pull request by cristian-ambrosini-sonarsource**
S1244 CSharp:
- RSPEC cleanup
- Promote rule to `SonarWay` profile

### CODE REVIEW

File: rules/S1244/csharp/rule.adoc [hunk 1]
```diff
@@ -1,59 +1,66 @@
 == Why is this an issue?
 
-Floating point math is imprecise because of the challenges of storing such values in a binary representation. Even worse, floating point math is not associative; push a ``++float++`` or a ``++double++`` through a series of simple mathematical operations and the answer will be different based on the order of those operation because of the rounding that takes place at each step.
-
+Floating point numbers are inherently imprecise due to the limitations of binary representation. 
+The non-associative nature of floating point arithmetic compounds this issue. The order in which operations are performed can affect the outcome due to the rounding that occurs at each step. This means that a series of mathematical operations can yield different results depending on the sequence of operations.
```
Comment (sebastien-marichal) [inline rules/S1244/csharp/rule.adoc:4]: > The non-associative nature of floating point arithmetic compounds this issue.

I find this sentence hard to understand.
I would add the link to wiki page about `associative`.

**Review by sebastien-marichal — CHANGES_REQUESTED** [review]
I find the description hard to understand.
I suggest adding some links and if possible reformulate it a bit.

File: rules/S1244/csharp/rule.adoc (hunk 1 above)
Comment (cristian-ambrosini-sonarsource) [reply rules/S1244/csharp/rule.adoc:4]: It's in the resources section, you mean adding it also here?

**Review by sebastien-marichal — APPROVED** [review]
LGTM!

**Pull request merged**